src/java.base/share/classes/java/lang/Class.java

```diff
@@ -4824,4 +4787,84 @@ private int getClassAccessFlagsRaw() {
     }
 
     private native int getClassAccessFlagsRaw0();
+
+    private Method findMethod(boolean publicOnly, String name, Class<?>... parameterTypes) {
+        PublicMethods.MethodList res = getMethodsRecursive(name, parameterTypes, true, publicOnly);
+        return res == null ? null : getReflectionFactory().copyMethod(res.getMostSpecific());
+    }
+
+    /**
+     * Return the first method that meets the requirements of an application main method
+     * {@jls 12.1.4}. The method must:
+     * <ul>
+     * <li>be declared in this class's hierarchy</li>
+     * <li>have the name "main"</li>
+     * <li>have a single argument of type {@code String[]}, {@code String...} or no argument</li>
+     * <li>have the return type of void</li>
+     * <li>be public, protected or package private</li>
+     * <li>not be abstract</li>
+     *</ul>
+     *
+     * Searching continues until a main method is found or the search is exhausted. The
+     * primary search occurs in two phases, once for a main method with a {@code
+     * String[]} or {@code String...} argument and failing that, once for a main method
+     * with a no arguments. The search itself uses recursion to first look at methods
+     * in this class, then default methods in this class's interface hierarchy and
+     * then repeating these steps with the class's super class.
+     *
+     * @apiNote The method returned may be declared in this class, a super class
+     * or as a default method of an interface that the class or super class
+     * implements.
+     * <p>It is not possible to declare a static main method and instance main
+     * method with the same signature in the same class. {@jls 8.4.2} states that
+     * "It is a compile-time error to declare two methods with override-equivalent
+     * signatures in a class."
+     * <p>{@link SecurityException SecurityExceptions} can halt
+     * the search. In this case, a null is returned.
```

Comment:
I see the latest update changes this to return null if denied by the SM. This seems very inconsistent with the other methods so I'm wondering why this has changed?

Reply:
Requested by CSR.